cmd/ddev/cmd/networks_test.go

```diff
@@ -0,0 +1,62 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/ddev/ddev/pkg/dockerutil"
+	docker "github.com/fsouza/go-dockerclient"
+	asrt "github.com/stretchr/testify/assert"
+)
+
+// TestNetworkDuplicates makes sure that Docker network duplicates
+// are deleted successfully with DDEV
+// See https://github.com/ddev/ddev/pull/5508
+// Note: duplicate networks cannot be created with Docker >=25.x.x
+// See https://github.com/moby/moby/pull/46251
+func TestNetworkDuplicates(t *testing.T) {
+	assert := asrt.New(t)
+
+	client := dockerutil.GetDockerClient()
+
+	// Create two networks with the same name
+	networkName := "ddev-" + t.Name() + "_default"
+
+	t.Cleanup(func() {
+		err := dockerutil.RemoveNetwork(networkName)
+		assert.NoError(err)
+
+		networks, err := client.ListNetworks()
+		assert.NoError(err)
+
+		// Ensure the network is not in the list
+		for _, network := range networks {
+			assert.NotEqual(networkName, network.Name)
+		}
+	})
+
+	labels := map[string]string{"com.ddev.platform": "ddev"}
+	netOptions := docker.CreateNetworkOptions{
+		Name:     networkName,
+		Driver:   "bridge",
+		Internal: false,
+		Labels:   labels,
+	}
+
+	// Create the first network
+	_, err := client.CreateNetwork(netOptions)
+	assert.NoError(err)
+
+	// Create a second network with the same name
+	_, err = client.CreateNetwork(netOptions)
+	if err != nil {
+		t.Logf("This test is run with Docker >=25.x.x, making it unrepresentative: %v", err)
+	}
```

Comment:
I'm not sure how to properly check for a duplicate here.

If I add `assert.NoError(err)`, it works for Docker 24, but not for Docker 25:
```
API error (409): network with name ddev-TestNetworkDuplicates_default already exists
```

So I can leave it as it is or add a strict `assert.NoError(err)` check and then the test will fail on every runner where Docker 25 is installed.

Reply:
Would it be worth checking for docker server version? Shouldn't be hard to do that using the api.

Reply:
That's what I should be doing here. Thank you!